contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -945,7 +945,9 @@ contract StakeManager is
             }
         }
 
-        validators[validatorId].initialRewardPerStake = newRewardPerStake;
+        if (newRewardPerStake > initialRewardPerStake) {
```

Comment:
This is in case of slashing or normal scenario also?

Reply:
Rewards should only grow, slashing works by slashing shares.